### PR TITLE
update release workflow to support macos arm64

### LINKED ISSUE #32: Build less things in PRs
We probably don't need to build for every arch under the sun in every PR. Probably just one linux one mac one windows.

### PR DESCRIPTION
**Pull request by kwigley**
resolves #32 and https://github.com/dbt-labs/dbt-core/issues/4185

- separate the release workflow from the test/lint workflow
- ensured we are build universal2 wheel files that support mac arm arch for supported python versions

Note: We can kick off a release for 0.4.0 again, it will publish the missing wheels.

### CODE REVIEW

**Comment by nathaniel-may** [comment]
I just noticed [here](https://pypi.org/project/dbt-extractor/#files) that we are only building arm64 for python 3.6 which is odd.

**Comment by kwigley** [comment]
Confirmed arm64 compat wheels are being generated for py3.6-py3.9

https://github.com/dbt-labs/dbt-extractor/runs/4411916006?check_suite_focus=true#step:7:8

**Comment by kwigley** [comment]
This is ready to go! you can see a list of distributions this workflow creates here https://github.com/dbt-labs/dbt-extractor/runs/4434445906?check_suite_focus=true#step:3:1

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+        uses: actions/upload-artifact@v2
+        with:
+          name: wheels
+          path: dist
+
+  python-linux:
+    name: Build linux ${{ matrix.target }}
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
```
Comment (nathaniel-may) [inline .github/workflows/release.yml:73]: why don't we need python version here?

**Review by nathaniel-may — APPROVED** [review]
lgtm. just one question.

File: .github/workflows/release.yml (hunk 1 above)
Comment (kwigley) [reply .github/workflows/release.yml:73]: When setting this argument
https://github.com/dbt-labs/dbt-extractor/blob/b6efb4784e9129b3ca5b2f1ea4478c1708de1d2b/.github/workflows/release.yml#L84
The `messense/maturin-action@v1` action builds wheels in a docker container and uses the python versions that are available in the docker container. You can see what python versions are supported here https://github.com/messense/maturin-action/blob/51478586be5dfb16a569cb6bbec182fba9f13f79/src/index.ts#L245

I'll add a comment to the source code calling this out!

**Pull request merged**